acctz/acctz.proto

```diff
@@ -99,34 +99,30 @@ message SessionInfo {
 // Details of authentication - for gRPC or shell/vendor-CLI.
 message AuthDetail {
   // Identity string is used to identify the user that executed the
-  // command. For instance, it could be the spiffe-id in the case of
+  // command. For instance, it could be the SPIFFE-ID in the case of
   // gRPC or unix-style user-name in the case of shell/vendor-CLI.
   string identity = 1;
 
-  // Privilege level configured on the system.
-  uint32 privilege_level = 2;
-
-  // authentication status
+  // Authentication status
   enum AuthenStatus {
     AUTHEN_STATUS_UNSPECIFIED = 0;
     AUTHEN_STATUS_PERMIT = 1;
     AUTHEN_STATUS_DENY = 2;
   }
-  AuthenStatus status = 3;
+  AuthenStatus status = 2;
 
   // In case of STATUS_DENY, cause for the deny
-  string deny_cause = 4;
+  string deny_cause = 3;
 }
 
 // Command details for shell/vendor-CLI
 message CommandService {
```

Comment:
We have a split of gRPC and "command" services however the command services listed here are a mix of separate APIs and each API behavior may or may not be able to be broken down into the constructs of commands and command arguments.

For CLI, it might be natural to break this down in such a fashion

However for other NBI APIs you have a mix of protocol level semantics coupled with data payloads in various structures, some of this is analogous to gRPC services where currently the plan is to encapsulate the entire message in a PROTO_ANY

I think this will need some additional consideration if we were to elaborate with examples on all services.  Currently in JUNOS, we log/record across various APIs by mostly breaking it down at the RPC name level only but not with full visibility into the remainder of request.

The full requests are analogous to debug/tracing which come with additional cost

Reply:
You seem to be conflating expanding the list service_requests with this log message?

I do not know what NBI is.

I do not see how the junos behavior you describe is different from the GrpcService message.

Reply:
You are suggesting splitting netconf/restconf in its own service_request type, because they more closely resemble (g)rpc and their arguments might not be representable in the string of cmd_args.  That seems reasonable to me.

For http, what I've seen of these interfaces, they are just conduit of sorts to the cli.  I realize that might not be universal and it could carry binary request data.  I suppose it co-exist with netconf, or be its own type.  Binary data could also be base64; which seems common.

I  do not know what any of this has to do with the deny_cause field. :)

Reply:
The comment is tagged to the `message CommandService` line but review rendering here makes it look like its against the changed line - Since this is a PR, not every line can hold a comment but rather requires a change within proximity (I had considered opening this for discussion w/ 1 or more distinct issues but opted for 1 PR to open up a wider discussion).  `deny_cause` has nothing to do w/ this but rather just had a alignment in field numbers within proximity.

That aside, I do think each distinct API will need need further analysis that is likely to be conveyed as their own message structs otherwise an additional ruleset would need to be devised on top of how to appropriately map, what will not map, etc...

Since none of that is defined today, it's just an additional point that there are versions of gNSI being cut (currently tagged as `1.2.2` at time of this comment) and implementation underway where there is too much ambiguity which is likely to create a significant amount of experimental backwards incompatible churn.  My suggestion is to eliminate such cases (removal) and build back up as time needs to be spent up front on the dissection of problem statements and specification vs. IDL.

Reply:
The purpose of 'commandservice' is actually to account for 'chris uses ssh/telnet to connect to a  device and types things'.

In a world where folk use some gNMI for config management (for instance) and gNOI for 'collection of stats/etc' MOST interaction is likely robots not humans. Where a human may still connect: "why is this bgp session not coming up?" "Why did this interface not do what I wanted?" "Oh, great, vendor bug, now I have to type 12 'test crash' type commands :("

we want to have the accounting data for that come over the gNSI path.